CalculateCurrentIssues.cls

```diff
@@ -1,71 +1,71 @@
 
-for (AggregateResult pub_list : find_publications) {
- pub_names.add((string)pub_list.get('publication_name__c'));
+for (AggregateResult pubList : findPublications) {
+ pubNames.add((string)pubList.get('publication_name__c'));
 }
 
-system.debug(pub_names);
+system.debug(pubNames);
 
-for (string this_pub : pub_names ) {
+for (string thisPub : pubNames ) {
 
- list<issue__c> all_pub_new_issues = [select id
+ list<issue__c> allPubNewIssues = [select id
                                  from issue__c
-                                 where publication_name__c = :this_pub
+                                 where publication_name__c = :thisPub
                                  and deadline__c >= :date.today()
                                  order by deadline__c];
 
- all_pub_new_issues[0].issue_status__c = 'Current';
- if (all_pub_new_issues.size()>1) {all_pub_new_issues[1].issue_status__c = 'Next';}
- if (all_pub_new_issues.size()>2) {all_pub_new_issues[2].issue_status__c = 'Next+1';}
- if (all_pub_new_issues.size()>3) {all_pub_new_issues[3].issue_status__c = 'Next+2';}
- if (all_pub_new_issues.size()>4) {all_pub_new_issues[4].issue_status__c = 'Next+3';}
- if (all_pub_new_issues.size()>5) {all_pub_new_issues[5].issue_status__c = 'Next+4';}
- if (all_pub_new_issues.size()>7) {all_pub_new_issues[6].issue_status__c = 'Next+5';}
- if (all_pub_new_issues.size()>8) {all_pub_new_issues[7].issue_status__c = 'Next+6';}
- if (all_pub_new_issues.size()>9) {all_pub_new_issues[8].issue_status__c = 'Next+7';}
- if (all_pub_new_issues.size()>10) {all_pub_new_issues[9].issue_status__c = 'Next+8';}
- if (all_pub_new_issues.size()>11) {all_pub_new_issues[10].issue_status__c = 'Next+9';}
- if (all_pub_new_issues.size()>12) {all_pub_new_issues[11].issue_status__c = 'Next+10';}
- if (all_pub_new_issues.size()>13) {all_pub_new_issues[12].issue_status__c = 'Next+11';}
+ allPubNewIssues[0].issue_status__c = 'Current';
+ if (allPubNewIssues.size()>1) {allPubNewIssues[1].issue_status__c = 'Next';}
+ if (allPubNewIssues.size()>2) {allPubNewIssues[2].issue_status__c = 'Next+1';}
+ if (allPubNewIssues.size()>3) {allPubNewIssues[3].issue_status__c = 'Next+2';}
+ if (allPubNewIssues.size()>4) {allPubNewIssues[4].issue_status__c = 'Next+3';}
+ if (allPubNewIssues.size()>5) {allPubNewIssues[5].issue_status__c = 'Next+4';}
+ if (allPubNewIssues.size()>7) {allPubNewIssues[6].issue_status__c = 'Next+5';}
+ if (allPubNewIssues.size()>8) {allPubNewIssues[7].issue_status__c = 'Next+6';}
+ if (allPubNewIssues.size()>9) {allPubNewIssues[8].issue_status__c = 'Next+7';}
+ if (allPubNewIssues.size()>10) {allPubNewIssues[9].issue_status__c = 'Next+8';}
+ if (allPubNewIssues.size()>11) {allPubNewIssues[10].issue_status__c = 'Next+9';}
+ if (allPubNewIssues.size()>12) {allPubNewIssues[11].issue_status__c = 'Next+10';}
+ if (allPubNewIssues.size()>13) {allPubNewIssues[12].issue_status__c = 'Next+11';}
  
- update all_pub_new_issues;
+ update allPubNewIssues;
```

Comment:
* This DML statement is inside a loop and is highly likely to hit governor limits. Please use collections to move the DML statement outside of the loop.
* This DML statement is not surrounded by any try-catch block. Please add one to ensure exceptions are handled properly.